frontend/wallet/src/main/DaemonProcess.re

```diff
@@ -27,70 +28,77 @@ let codaCommand = (~port, ~extraArgs) => {
         |],
       ),
     env:
-      Some(
-        Js.Dict.fromList([
-          (
-            "CODA_KADEMLIA_PATH",
-            ProjectRoot.resource ^/ codaPath ^/ "kademlia",
-          ),
-        ]),
+      Js.Dict.fromArray(
+        Js.Array.concat(
+          [|
+            (
+              "CODA_KADEMLIA_PATH",
+              ProjectRoot.resource ^/ codaPath ^/ "kademlia",
+            ),
+          |],
+          Js.Dict.entries(ChildProcess.Process.env),
+        ),
       ),
+    logfileName: "daemon",
   };
 };
 
 module Process = {
   let start = (command: Command.t) => {
-    let {Command.executable, args} = command;
-    print_endline({j|Starting $executable with $args|j});
+    let {Command.executable, args, logfileName} = command;
+    print_endline(
+      {j|Starting $executable with $args. Logging to `$logfileName.log`|j},
+    );
+
+    /*
+     child_process.spawn communicates with processes using sockets.
+     When the proposer process starts up in the daemon, stdout and stderr
+     are accessed using a helper function from async_unix:
+     https://github.com/janestreet/async_unix/blob/ff13d69c96f4b857737263910b3f6b08311b5dfc/src/writer0.ml#L1465
+     This causes async_unix to attempt to infer the type of stdout/err
+     using and check if it has SO_ACCEPTCONN set using getsockopt.
+     This flag is not supported on Mac, so it crashes with "Protocol not available"
+
+     As a workaround, we have to redirect all output to a file.
+
+     If this issue is ever resolved or we stop using async_unix, this code can be simplified:
+     https://github.com/janestreet/async_unix/issues/15
+     */
 
+    let log = Fs.openSync(command.logfileName ++ ".log", "w");
     let p =
```

Comment:
this might be a bit pedantic 😅 but could we rename this `process` (or if that's a protected keyword, `proposer` / `proposerProcess`?  Feel free to ignore this comment

Reply:
sure